obp/dataset/real.py

```diff
@@ -308,8 +334,18 @@ def sample_bootstrap_bandit_feedback(
                 test_size=test_size, is_timeseries_split=is_timeseries_split
             )
         n_rounds = bandit_feedback["n_rounds"]
+        if sample_size is None:
```

Comment:
[imo]
If we can postpone the type check of `sample_size`, it seems to be simple to remove the block of `if sample_size: 
 (L321)` and use `max_val` as follows:

```
if sample_size is None:
    sample_size = bandit_feedback["n_rounds"]
else:
    check_scalar(
        sample_size,
        name="sample_size",
        target_type=(int),
        min_val=0,
        max_val=n_rounds,
    )
```